clinvar/src/main/scala/org/broadinstitute/monster/etl/clinvar/models/intermediate/VariationArchive.scala

```diff
@@ -247,60 +239,95 @@ object VariationArchive {
           // NOTE: Because trait mappings link to the numeric ID for each SCV,
           // but we use the accession as the PK, we need to do a little bit
           // of post-processing on the mappings.
-          val scvId = rawScv
-            .extract("@ID")
-            .getOrElse {
-              throw new IllegalStateException(s"Found an SCV with no numeric ID: $scv")
-            }
-            .str
-          scvIdToAccession.update(scvId, scv.id)
+          val scvId = SCV.extractNumericId(rawScv)
+          scvIdToAccession.update(scvId, scvAccessionId)
           val relevantMappings = mappingsByScvId.getOrElse(scvId, Array.empty)
+          val clinicalAssertionTraitSetId = new mutable.ArrayBuffer[String]()
+          var relatedVcvTraitSetId: Option[String] = None
```

Comment:
In general, `var` is best to avoid. It's needed here to make the logic work with the existing `foreach`- and buffer-based control flow, but we can refactor to avoid it.

Instead of the existing:
```scala
rawScv.extract("TraitSet").foreach { rawTraitSet =>
  // code that pushed into `ArrayBuffer`s and sets `var`s
}
```
We can do:
```scala
val (directScvTraitSet, directScvTraits) = rawScv.extract("TraitSet").map { rawTraitSet =>
  // just the extraction part of the code
}
```
Then we have direct access to the pieces we need for VCVTraitSet-/RCV-linking, and we can still push into the top-level `ArrayBuffer`s outside of the `map`:
```scala
scvTraitSets.append(directScvTraitSet)
scvTraits.appendAll(directScvTraits)
```

Reply:
do we want to pull the `append`s out of the `rawScv.extract("TraitSet").map` because `map` shouldn't have side effects?

Reply:
Ideally, yes